### DISCUSSION TITLE
Error on linker.compare_two_records()

### DISCUSSION THREAD

**Discussion by mmagoffin-sd** [moj-analytical-services/splink#769] [Q&A] [answered]
Hello!
I've been trying to make use of the comparing two records feature, but I keep getting the following error: does anyone have any ideas what I'm doing wrong/ what to fix?
<img width="989" alt="Screen Shot 2022-09-13 at 2 43 06 PM" src="https://user-images.githubusercontent.com/85239124/189984656-47501cb4-541b-4c94-b334-441ddb021786.png">
<img width="562" alt="Screen Shot 2022-09-13 at 2 42 54 PM" src="https://user-images.githubusercontent.com/85239124/189984667-588bb09d-ddc5-4d19-b767-ad80ab85e107.png">

**Comment by RobinL** [comment, marked as answer]
Hello - I have been trying to track down the source of this bug.  I might have fixed it in #787.  Please could you upgrade to `pip install splink==3.3.3` and see whether you still get this error.

**Reply by mmagoffin-sd** [reply]
Hmm unfortunately I'm still getting the same error, however now instead of erroring on __splink_df_tf_fname it errors on __splink_df_tf_lname, only thing that I can think of that might effected it was I added a null level to my fname comparison levels? Both of those comparisons have an exact match level with term frequency adjustments.

Also, I now get the word "finally" printed multiple times when running linker.predict(), as seen in the picture below
<img width="869" alt="Screen Shot 2022-09-21 at 4 30 46 PM" src="https://user-images.githubusercontent.com/85239124/191604357-c893ea49-dc75-4349-9798-7e08e3c7d9aa.png">

**Reply by RobinL** [reply]
Sorry - just looked at this again and I think I know the source of the problem.  

It's a flaw in the API whereby you need to manually compute the term frequency tables for all columns with tf adjustments.  We really need to add a warning/other information to the user to tell them what's gone wrong.

For the moment, to fix this, you can see an example at the top of this (see 'step 1')
https://moj-analytical-services.github.io/splink/demos/example_real_time_record_linkage.html

i.e. you need lines like `linker.compute_tf_table("first_name")` prior to running `compare_two_records`


The problem is that the term frequency tables have not been computed.

Upgrade to the latest version to get rid of that 'finally' message, it was a print statement i left in by mistake

**Reply by mmagoffin-sd** [reply]
That worked, thank you!
